src/connection.js

```diff
@@ -986,6 +1023,7 @@ class Connection extends EventEmitter {
       }
 
       this.request = request;
+      this.request.connection = this;
```

Comment:
I don't understand. Request.pauseCallback/resumeCallback would be private but still can be accessed from within the Connection class. Why can't we make Request.connection private and access it from within the Connection class and make Connection.pauseRequest()/resumeRequest() private and access it from within the Request class?